### PR TITLE
Clarify the auth process

### PR DESCRIPTION
**Pull request by zaiste**

### CODE REVIEW

**Review by 2can — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -70,15 +70,58 @@ export default function App({ Component, pageProps }: AppProps) {
...
-  email: "example@mail.com",
-  passowrd: "password",
-});
+export const LoginPage = () => {
+  const { signIn, signOut, isAuthenticating } = useSaleorAuthContext();
+
+  const [{ data: currentUser, loading: isLoadingCurrentUser }] = useQuery(CurrentUserDocument);
+
+  // important: wait for both the end of auth process and the end of query invocation
+  const isLoading = isAuthenticating || isLoadingCurrentUser;
```
Comment (mmiszy) [inline README.md:85]: Are you sure it's necessary to wait for `isAuthenticating`? I believe it's implicit and we can simplify these examples.

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -70,15 +70,58 @@ export default function App({ Component, pageProps }: AppProps) {
...
+
+  const submitHandler = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const result = await signIn({ email: "example@mail.com", passowrd: "password" });
```
Comment (mmiszy) [inline README.md:90]:
```suggestion
    const result = await signIn({ email: "example@mail.com", password: "password" });
```

File: README.md (hunk 1 above)
Comment (mmiszy) [reply README.md:85]: Why? It doesn't affect the way this example works – it'd be the same with or without `isAuthenticating`. IMO it's making the example more complicated than it needs to be.

Comment (mmiszy) [reply README.md:85]: In other words: `isAuthenticating || isLoadingCurrentUser === isLoadingCurrentUser`

**Pull request merged**